lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -518,9 +518,13 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
             final Map<String, dynamic> arguments = decoded.arguments as Map<String, dynamic>;
             // TODO(ferhat): Find more appropriate defaults? Or noop when values are null?
             final String label = arguments['label'] as String? ?? '';
-            final int primaryColor = arguments['primaryColor'] as int? ?? 0xFF000000;
             domDocument.title = label;
-            setThemeColor(ui.Color(primaryColor));
+            replyToPlatformMessage(callback, codec.encodeSuccessEnvelope(true));
+            return;
+          case 'SystemChrome.setSystemUIOverlayStyle':
```

Comment:
According to the API docs:

> Specifies a preference for the style of the system overlays.
> * `statusBarBrightness → Brightness?` The brightness of top status bar.
> * **`statusBarColor → Color?` The color of top status bar.**
> * `statusBarIconBrightness → Brightness?` The brightness of the top status bar icons.
> * `systemNavigationBarColor → Color?` The color of the system bottom navigation bar.
> ...

I agree this seems like a more correct place to set the `statusBarColor`.